crates/filter/src/dynamic.rs

```diff
@@ -330,12 +335,16 @@ macro_rules! source_iter_file {
 // Generate an filtered iterator from Source::List(list).
 macro_rules! source_iter_list {
     ( $scorer:ident, $list:ident ) => {
-        $list.filter_map(|line| $scorer(&line).map(|(score, indices)| (line, score, indices)))
+        $list.filter_map(|item| {
+            // XXX: How do I avoid cloning here?
+            // let line = item.filter.clone().unwrap_or(item.display.clone());
+            $scorer(item)
+        })
     };
 }
```

Comment:
```rust
macro_rules! source_iter_list {
    ( $scorer:expr, $list:expr ) => {
        match ($list, $scorer) {
            (list, scorer) => list.filter_map(|item| {
                let line = item.filter.unwrap_or(item.display);
                scorer(&line).map(|(score, indices)| (line, score, indices))
            })
        }
    };
}
```

If you need to preserve the item instead of line, then it would be

```rust
macro_rules! source_iter_list {
    ( $scorer:expr, $list:expr ) => {
        match ($list, $scorer) {
            (list, scorer) => list.filter_map(|item| {
                scorer(item.filter.as_deref().unwrap_or(&item.display)).map(|(score, indices)| (item, score, indices))
            })
        }
    };
}
```

Reply:
Good to know! It's a left-over of an initial transformation but I did run into that problem during that initial transformation.